src/ViewerOptions.tsx

```diff
@@ -0,0 +1,199 @@
+import React from "react";
+import { PluginContext } from "molstar/lib/mol-plugin/context";
+import { Script } from "molstar/lib/mol-script/script";
+import { StructureSelection } from "molstar/lib/mol-model/structure";
+import { StructureSelectionQuery } from "molstar/lib/mol-plugin-state/helpers/structure-selection-query";
+import { setStructureOverpaint } from "molstar/lib/mol-plugin-state/helpers/structure-overpaint";
+import { MolScriptBuilder as MS } from "molstar/lib/mol-script/language/builder";
+import { Structure } from "molstar/lib/mol-model/structure";
+import { Color } from "molstar/lib/mol-util/color";
+
+interface ViewerOptionsProps {
+    plugin: React.MutableRefObject<PluginContext | null>;
+}
+
+const ViewerOptions: React.FunctionComponent<ViewerOptionsProps> = ({
+    plugin,
+}) => {
+    async function createComponent(
+        event: React.SyntheticEvent<HTMLFormElement>
+    ) {
+        event.preventDefault();
+        const form = event.currentTarget;
+        const formElements = form.elements as typeof form.elements & {
+            start: { value: number };
+            end: { value: number };
+            // colourPicker: { value: string };
+            name: { value: string };
+            style: { value: string };
+        };
+        const range = Array.from(
+            { length: formElements.end.value - formElements.start.value },
+            (v, k) => k + Number(formElements.start.value)
```

Comment:
What does this functionally mean, index + the start value?
Also might be worth converting the start and end values to ints before the loop. You're relying on the minus operator to convert in the length, but the + operator is overloaded:
```js
1 + 1 // 2
1 + '1' // '11'
'1' - 1 // 0
```

Probably use `parseInt` or `parseFloat` instead of initialising `Number` class.

```suggestion
            (_, idx) => idx + Number(formElements.start.value)
```

Reply:
Converted values before loop and added comment